### PR TITLE
Egnyte connector

### PR DESCRIPTION
**Pull request by Weves**
Also adds a new "standard" approach to adding oauth based connectors.

### CODE REVIEW

File: backend/danswer/server/documents/standard_oauth.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    base_url = str(request.base_url)
+    if "127.0.0.1" in base_url:
+        base_url = base_url.replace("127.0.0.1", "localhost")
+
+    # store state in redis
+    if not desired_return_url:
+        desired_return_url = f"{WEB_DOMAIN}/admin/connectors/{source}?step=0"
+    redis_client = get_redis_client(tenant_id=tenant_id)
+    state = str(uuid.uuid4())
+    redis_client.set(_OAUTH_STATE_KEY_FMT.format(state=state), desired_return_url)
```
Comment (Weves) [inline backend/danswer/server/documents/standard_oauth.py]: add expiration

File: backend/danswer/connectors/egnyte/connector.py [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    data: dict[str, Any] | None = None,
+    headers: dict[str, Any] | None = None,
+    params: dict[str, Any] | None = None,
+    timeout: int = _TIMEOUT,
+    stream: bool = False,
+    tries: int = 8,
+    delay: float = 1,
+    backoff: float = 2,
+) -> requests.Response:
+    @retry(tries=tries, delay=delay, backoff=backoff, logger=cast(Logger, logger))
```
Comment (rkuo-danswer) [inline backend/danswer/connectors/egnyte/connector.py:57]: couple of possible alternative approaches to think about in the future:
- set urllib3.Retry on the requests object
- tenacity Retry is better supported nowadays

File: backend/danswer/connectors/egnyte/connector.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    """
+    # Skip if file path doesn't match folder path filter
+    if folder_path and not file_metadata["path"].startswith(folder_path):
+        raise ValueError(
+            f"File path {file_metadata['path']} does not match folder path {folder_path}"
+        )
+
+    file_name = file_metadata["name"]
+    extension = get_file_ext(file_name)
+    if not check_file_ext_is_valid(extension):
```
Comment (rkuo-danswer) [inline backend/danswer/connectors/egnyte/connector.py]: nit: function naming is a bit redundant ... could be is_valid_file_ext

File: backend/danswer/connectors/egnyte/connector.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        self.folder_path = folder_path or ""  # Root folder if not specified
+        self.batch_size = batch_size
+        self.access_token: str | None = None
+
+    @classmethod
+    def oauth_id(cls) -> DocumentSource:
+        return DocumentSource.EGNYTE
+
+    @classmethod
+    def redirect_uri(cls, base_domain: str, state: str) -> str:
```
Comment (rkuo-danswer) [inline backend/danswer/connectors/egnyte/connector.py]: this method is somewhat confusingly named ... the url being generated is the authorization url for the integration.

File: backend/danswer/connectors/egnyte/connector.py [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+
+                if doc is not None:
+                    current_batch.append(doc)
+
+                    if len(current_batch) >= self.batch_size:
+                        yield current_batch
+                        current_batch = []
+
+            except Exception as e:
+                logger.error(f"Failed to process file {file['path']}: {str(e)}")
```
Comment (rkuo-danswer) [inline backend/danswer/connectors/egnyte/connector.py]: logger.exception usually easier to use (implicitly logs exceptions)

**Review by rkuo-danswer — APPROVED** [review]

**Pull request merged**